### ISSUE TITLE
Do not pad highest dimension of memory

### ISSUE THREAD

**Issue by mattfel1** [stanford-ppl/spatial#241]
Padding is only needed to keep bank/ofs resolution consistent when higher dimensions increment.  Having padding on the most significant dimension is pointless and confusing

**Comment by mattfel1** [comment]
Actually this was a bad idea because it throws off the padding calculator that tries to minimize area.  For example, a mem my have N = 21, dims = 7,25, alpha = 1,7.  Two of its P options are 21,1 and 7,3.  It SHOULD choose 7,3 since it results in a pad of 0,2 and therefore there are ceil(7 * (25+2)/21) = 9 elements per bank.  But if we force no-pad for the highest dim, the P option of 21,1 results in each bank having ceil(7*25/21)=9 elements per bank, but address 0,24 of this memory should have ofs=24, which wasn't accounted for because we ignored the padding.  

TLDR padding the highest dimension seems unnecessary from the point of view of the bank/ofs calculation arithmetic, but is actually required when sizing the physical BRAMs and assigning the correct number of address bits to them
